nlp/optimize/tron.py

```diff
@@ -265,25 +265,21 @@ def projected_newton_step(self, x, g, H, delta, l, u, s, cgtol, itermax):
                               radius=self.tr.radius,
                               abstol=tol)
 
-            step = self.solver.step
+            step = np.zeros(x.size)
+            step[free_vars] = self.solver.step
             iters += self.solver.niter
 
-            # Exit if the solver took no additional steps
+            # Exit immediately if the solver took no additional steps
             if self.solver.niter == 0:
                 exitOptimal = True
                 info = 4
+                break
 
             # Use a projected search to obtain the next iterate
-            (xfree, proj_step) = self.projected_linesearch(x[free_vars],
-                                                           l[free_vars],
-                                                           u[free_vars],
-                                                           gfree, step, ZHZ,
-                                                           alpha=1.0)
+            (x, s) = self.projected_linesearch(x, l, u, g, s,
+                                               step, H, alpha=1.0)
```

Comment:
What was the problem here? The idea of active-set methods is to solve a smaller problem, not a full-dimensional problem.